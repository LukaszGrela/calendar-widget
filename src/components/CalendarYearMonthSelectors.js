import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const noop = () => { };
const CalendarYearMonthSelectors = (props) => {
    const { className, month, year, monthChanged = noop, yearChanged = noop } = props;
    const monthList = moment.monthsShort();
    const selectedMonth = monthList[month];
    const yearList = [year - 10, year - 9, year - 8, year - 7, year - 6, year - 5, year - 4, year - 3, year - 2, year - 1,
        year,
    year + 1, year + 2, year + 3, year + 4, year + 5, year + 6, year + 7, year + 8, year + 9, year + 10];
    return (
        <div className={`year-month-selectors ${className}`}>
            <select
                className='month-selector'
                name='month-selector'
                onChange={(e) => monthChanged(e.target.value)}
                value={selectedMonth}>
                {
                    monthList.map((month, index) => <option key={index} value={month}>{month}</option>)
                }
            </select>
            <select
                className='year-selector'
                name='year-selector'
                onChange={(e) => yearChanged(e.target.value)}
                value={year}>
                {
                    yearList.map((year, index) => <option key={index} value={year}>{year}</option>)
                }
            </select>
        </div>
    )
}
CalendarYearMonthSelectors.propTypes = {
    year: PropTypes.number.isRequired,
    month: PropTypes.number.isRequired,
    monthChanged: PropTypes.func,
    yearChanged: PropTypes.func
};

export default CalendarYearMonthSelectors;