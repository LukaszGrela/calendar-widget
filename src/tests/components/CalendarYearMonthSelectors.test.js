import React from 'react';
import { shallow } from 'enzyme';
import CalendarYearMonthSelectors from '../../components/CalendarYearMonthSelectors';


let wrapper, year, month, monthChanged, yearChanged;
jest.unmock('moment');
beforeEach(() => {
    year = 1979;
    month = 5;
    monthChanged = jest.fn();
    yearChanged = jest.fn();
    wrapper = shallow(<CalendarYearMonthSelectors
        year={year}
        month={month}
    />);
});

test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call monthChanged', () => {

    wrapper.setProps({
        monthChanged: monthChanged
    });

    const month = wrapper.find('.month-selector');
    expect(month).toHaveLength(1);

    month.at(0).simulate('change', { target: { value: 'Jan' } });
    expect(monthChanged).toHaveBeenLastCalledWith('Jan');
});

test('Should call yearChanged', () => {

    wrapper.setProps({
        yearChanged: yearChanged
    });

    const month = wrapper.find('.year-selector');
    expect(month).toHaveLength(1);

    month.at(0).simulate('change', { target: { value: '1980' } });
    expect(yearChanged).toHaveBeenLastCalledWith('1980');
});
