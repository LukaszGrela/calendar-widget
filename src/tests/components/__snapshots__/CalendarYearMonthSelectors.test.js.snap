// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="year-month-selectors undefined"
>
  <select
    className="month-selector"
    name="month-selector"
    onChange={[Function]}
    value="Jun"
  >
    <option
      key="0"
      value="Jan"
    >
      Jan
    </option>
    <option
      key="1"
      value="Feb"
    >
      Feb
    </option>
    <option
      key="2"
      value="Mar"
    >
      Mar
    </option>
    <option
      key="3"
      value="Apr"
    >
      Apr
    </option>
    <option
      key="4"
      value="May"
    >
      May
    </option>
    <option
      key="5"
      value="Jun"
    >
      Jun
    </option>
    <option
      key="6"
      value="Jul"
    >
      Jul
    </option>
    <option
      key="7"
      value="Aug"
    >
      Aug
    </option>
    <option
      key="8"
      value="Sep"
    >
      Sep
    </option>
    <option
      key="9"
      value="Oct"
    >
      Oct
    </option>
    <option
      key="10"
      value="Nov"
    >
      Nov
    </option>
    <option
      key="11"
      value="Dec"
    >
      Dec
    </option>
  </select>
  <select
    className="year-selector"
    name="year-selector"
    onChange={[Function]}
    value={1979}
  >
    <option
      key="0"
      value={1929}
    >
      1929
    </option>
    <option
      key="1"
      value={1930}
    >
      1930
    </option>
    <option
      key="2"
      value={1931}
    >
      1931
    </option>
    <option
      key="3"
      value={1932}
    >
      1932
    </option>
    <option
      key="4"
      value={1933}
    >
      1933
    </option>
    <option
      key="5"
      value={1934}
    >
      1934
    </option>
    <option
      key="6"
      value={1935}
    >
      1935
    </option>
    <option
      key="7"
      value={1936}
    >
      1936
    </option>
    <option
      key="8"
      value={1937}
    >
      1937
    </option>
    <option
      key="9"
      value={1938}
    >
      1938
    </option>
    <option
      key="10"
      value={1939}
    >
      1939
    </option>
    <option
      key="11"
      value={1940}
    >
      1940
    </option>
    <option
      key="12"
      value={1941}
    >
      1941
    </option>
    <option
      key="13"
      value={1942}
    >
      1942
    </option>
    <option
      key="14"
      value={1943}
    >
      1943
    </option>
    <option
      key="15"
      value={1944}
    >
      1944
    </option>
    <option
      key="16"
      value={1945}
    >
      1945
    </option>
    <option
      key="17"
      value={1946}
    >
      1946
    </option>
    <option
      key="18"
      value={1947}
    >
      1947
    </option>
    <option
      key="19"
      value={1948}
    >
      1948
    </option>
    <option
      key="20"
      value={1949}
    >
      1949
    </option>
    <option
      key="21"
      value={1950}
    >
      1950
    </option>
    <option
      key="22"
      value={1951}
    >
      1951
    </option>
    <option
      key="23"
      value={1952}
    >
      1952
    </option>
    <option
      key="24"
      value={1953}
    >
      1953
    </option>
    <option
      key="25"
      value={1954}
    >
      1954
    </option>
    <option
      key="26"
      value={1955}
    >
      1955
    </option>
    <option
      key="27"
      value={1956}
    >
      1956
    </option>
    <option
      key="28"
      value={1957}
    >
      1957
    </option>
    <option
      key="29"
      value={1958}
    >
      1958
    </option>
    <option
      key="30"
      value={1959}
    >
      1959
    </option>
    <option
      key="31"
      value={1960}
    >
      1960
    </option>
    <option
      key="32"
      value={1961}
    >
      1961
    </option>
    <option
      key="33"
      value={1962}
    >
      1962
    </option>
    <option
      key="34"
      value={1963}
    >
      1963
    </option>
    <option
      key="35"
      value={1964}
    >
      1964
    </option>
    <option
      key="36"
      value={1965}
    >
      1965
    </option>
    <option
      key="37"
      value={1966}
    >
      1966
    </option>
    <option
      key="38"
      value={1967}
    >
      1967
    </option>
    <option
      key="39"
      value={1968}
    >
      1968
    </option>
    <option
      key="40"
      value={1969}
    >
      1969
    </option>
    <option
      key="41"
      value={1970}
    >
      1970
    </option>
    <option
      key="42"
      value={1971}
    >
      1971
    </option>
    <option
      key="43"
      value={1972}
    >
      1972
    </option>
    <option
      key="44"
      value={1973}
    >
      1973
    </option>
    <option
      key="45"
      value={1974}
    >
      1974
    </option>
    <option
      key="46"
      value={1975}
    >
      1975
    </option>
    <option
      key="47"
      value={1976}
    >
      1976
    </option>
    <option
      key="48"
      value={1977}
    >
      1977
    </option>
    <option
      key="49"
      value={1978}
    >
      1978
    </option>
    <option
      key="50"
      value={1979}
    >
      1979
    </option>
    <option
      key="51"
      value={1980}
    >
      1980
    </option>
    <option
      key="52"
      value={1981}
    >
      1981
    </option>
    <option
      key="53"
      value={1982}
    >
      1982
    </option>
    <option
      key="54"
      value={1983}
    >
      1983
    </option>
    <option
      key="55"
      value={1984}
    >
      1984
    </option>
    <option
      key="56"
      value={1985}
    >
      1985
    </option>
    <option
      key="57"
      value={1986}
    >
      1986
    </option>
    <option
      key="58"
      value={1987}
    >
      1987
    </option>
    <option
      key="59"
      value={1988}
    >
      1988
    </option>
    <option
      key="60"
      value={1989}
    >
      1989
    </option>
    <option
      key="61"
      value={1990}
    >
      1990
    </option>
    <option
      key="62"
      value={1991}
    >
      1991
    </option>
    <option
      key="63"
      value={1992}
    >
      1992
    </option>
    <option
      key="64"
      value={1993}
    >
      1993
    </option>
    <option
      key="65"
      value={1994}
    >
      1994
    </option>
    <option
      key="66"
      value={1995}
    >
      1995
    </option>
    <option
      key="67"
      value={1996}
    >
      1996
    </option>
    <option
      key="68"
      value={1997}
    >
      1997
    </option>
    <option
      key="69"
      value={1998}
    >
      1998
    </option>
    <option
      key="70"
      value={1999}
    >
      1999
    </option>
    <option
      key="71"
      value={2000}
    >
      2000
    </option>
    <option
      key="72"
      value={2001}
    >
      2001
    </option>
    <option
      key="73"
      value={2002}
    >
      2002
    </option>
    <option
      key="74"
      value={2003}
    >
      2003
    </option>
    <option
      key="75"
      value={2004}
    >
      2004
    </option>
    <option
      key="76"
      value={2005}
    >
      2005
    </option>
    <option
      key="77"
      value={2006}
    >
      2006
    </option>
    <option
      key="78"
      value={2007}
    >
      2007
    </option>
    <option
      key="79"
      value={2008}
    >
      2008
    </option>
    <option
      key="80"
      value={2009}
    >
      2009
    </option>
    <option
      key="81"
      value={2010}
    >
      2010
    </option>
    <option
      key="82"
      value={2011}
    >
      2011
    </option>
    <option
      key="83"
      value={2012}
    >
      2012
    </option>
    <option
      key="84"
      value={2013}
    >
      2013
    </option>
    <option
      key="85"
      value={2014}
    >
      2014
    </option>
    <option
      key="86"
      value={2015}
    >
      2015
    </option>
    <option
      key="87"
      value={2016}
    >
      2016
    </option>
    <option
      key="88"
      value={2017}
    >
      2017
    </option>
    <option
      key="89"
      value={2018}
    >
      2018
    </option>
    <option
      key="90"
      value={2019}
    >
      2019
    </option>
    <option
      key="91"
      value={2020}
    >
      2020
    </option>
    <option
      key="92"
      value={2021}
    >
      2021
    </option>
    <option
      key="93"
      value={2022}
    >
      2022
    </option>
    <option
      key="94"
      value={2023}
    >
      2023
    </option>
    <option
      key="95"
      value={2024}
    >
      2024
    </option>
    <option
      key="96"
      value={2025}
    >
      2025
    </option>
    <option
      key="97"
      value={2026}
    >
      2026
    </option>
    <option
      key="98"
      value={2027}
    >
      2027
    </option>
    <option
      key="99"
      value={2028}
    >
      2028
    </option>
    <option
      key="100"
      value={2029}
    >
      2029
    </option>
  </select>
</div>
`;
